### PR TITLE
Add note about descriptors and why they aren't supported.

### PR DESCRIPTION
**Pull request by thomasvl**

### CODE REVIEW

File: Documentation/API.md [hunk 1]
```diff
@@ -679,6 +679,24 @@ public protocol ExtensionMap {
 }
 ```
 
+## Descriptors
+
+Some other languages expose _Descriptor_ objects for messages, enums, fields,
+and oneof; but not all languages.  And the `.proto` language allows developers
```
Comment (allevato) [inline Documentation/API.md]: s/;/,/

s/And the \`.proto\` language/The \`.proto\` language also/

File: Documentation/API.md [hunk 2]
```diff
@@ -679,6 +679,24 @@ public protocol ExtensionMap {
...
+
+Support for descriptors ends up requiring some amount of code, but more
+importantly it requires capturing a large binary blob of data for every
+message, enum, oneof, etc. That data has two potenial issues, it bloats the
+binaries, and it is something that can be extracted from the binary to help
+reverse engineer details about the binary.
+
+For these reasons, SwiftProtobuf does not current support anything like the
+Descriptor objects. It is something that could get revisited in the future,
+but will need careful consideration, the bloat/size issues is of the most
```
Comment (allevato) [inline Documentation/API.md]: s/,/;/

File: Documentation/API.md [hunk 3]
```diff
@@ -679,6 +679,24 @@ public protocol ExtensionMap {
...
+concern because of Swifts common use for mobile applications.
```
Comment (allevato) [inline Documentation/API.md]: s/Swifts/Swift's/

File: Documentation/API.md (hunk 2 above)
Comment (thomasvl) [reply Documentation/API.md]: Done

File: Documentation/API.md (hunk 3 above)
Comment (thomasvl) [reply Documentation/API.md]: Done.

File: Documentation/API.md (hunk 1 above)
Comment (thomasvl) [reply Documentation/API.md]: Done

**Pull request merged**